client/my-sites/stats/stats-purchase/index.tsx

```diff
@@ -54,6 +55,10 @@ const StatsPurchasePage = ( { query }: { query: { redirect_uri: string; from: st
 		return isProductOwned( siteProducts, PRODUCT_JETPACK_STATS_PWYW_YEARLY );
 	}, [ siteProducts ] );
 
+	if ( query?.from && query.from === 'jetpack-my-jetpack' ) {
```

Comment:
We've got `from` recorded with the page view of the page, so I guess we could omit this one?

https://github.com/Automattic/wp-calypso/blob/f3aedf8a17acc280bdfc9036d6b0a4b65434bbc9/client/my-sites/stats/stats-purchase/index.tsx#L86